uploader/uploader.py

```diff
@@ -40,7 +40,14 @@ def __init__(self):
                 raise RuntimeError('Won\'t let you upload to localhost because I '
                                    'remove files after uploading them, and you '
                                    'might be uploading to the same directory')
-            self.mode = 'rsync'
+        url_check = re.search(r'/[a-zA-Z0-9!\"£$%^&*()_+=-\[\];'#|,.\|{}:@~<>?`¬]+$', self.url)
```

Comment:
Why not require `self.url` to end with `/'?  That seems much simpler.

Reply:
Yep, it does.

Updated now.